lib/tests/use-union-only-when-multiple-choices.test.ts

```diff
@@ -18,5 +18,5 @@ test("use-union-only-when-multiple-choices", () => {
                 },
             },
         })
-    ).toMatchInlineSnapshot('"z.object({ singleOneOf: z.string(), multipleOneOf: z.union([z.string(), z.number()]), singleAnyOf: z.string(), multipleAnyOf: z.union([z.union([z.string(), z.number()]), z.array(z.union([z.string(), z.number()]))]), singleAllOf: z.string(), multipleAllOf: z.string().and(z.number()) }).partial()"');
+    ).toMatchInlineSnapshot('"z.object({ singleOneOf: z.string(), multipleOneOf: z.union([z.string(), z.number()]), singleAnyOf: z.string(), multipleAnyOf: z.union([z.string(), z.number()]), singleAllOf: z.string(), multipleAllOf: z.string().and(z.number()) }).partial()"');
```

Comment:
is this one still valid ? 
other than this change, LGTM we can probably merge it

Reply:
from what I remember and looking at [this comment](https://github.com/astahmer/openapi-zod-client/blob/main/lib/src/openApiToZod.ts#L103)

> // anyOf = oneOf but with 1 or more = `T extends oneOf ? T | T[] : never`

so when I see this: 

```json
{
   anyOf: [{ type: "string" }, { type: "number" }]
}
```

I think of it as:

```ts
type Thing = string | number | (Array<string | number>)
```

which looked kinda ok before:
```diff
- z.union([z.union([z.string(), z.number()]), z.array(z.union([z.string(), z.number()]))])
+ z.union([z.string(), z.number()])
```

and now it looks like :

```ts
type Thing = string | number
```

was it wrong before then or is it now ?

Reply:
i think it was wrong before. anyOf doesn't mean it can be an array. it is a union, that allows any additional properties through (it stops validating after the first type match).

it was kind of hard to find an authoritative source on this, so i looked at the tests from the json-schema repo here. they don't show any use of arrays https://github.com/json-schema-org/JSON-Schema-Test-Suite/blob/b069ac352c3dc1fae71f6f177dafe97001c97920/tests/draft2020-12/anyOf.json

i also tested this with `ajv` which is a popular json-schema implementation in js. see here, it fails (as expected) in the array case https://stackblitz.com/edit/json-schema-anyof-test-pnfj13?file=index.tsx%3AL55

Reply:
it would probably be super valuable at some point to lift all those json-schema test cases and run them against the generated zod schemas!

Reply:
ok got it, I checked with this guy and it explains it pretty clearly
https://chat.openai.com/share/954082b4-d72d-4b5f-a0d1-4a2107705ddb

oneOf = exactly 1 match, anyOf = must match 1 or more in the list

I guess we can merge it then if that's fine with you

Reply:
in my testing i don't think the "or more" part is even a requirement - as soon as there's a single match it doesn't care about the rest

```js
{
  anyOf: [
    {
      type: 'object',
      properties: {
        age: {
          type: 'integer',
        },
      },
      required: ['age'],
    },
    {
      type: 'object',
      properties: {
        pet_type: {
          type: 'string',
          enum: ['Cat', 'Dog'],
        },
      },
      required: ['pet_type'],
    },
  ],
};
```

this schema would allow the following data: 

```js
{ pet_type: "Cat", age: "not a number" } // matches PetByType, doesn't care age isn't a number
{ age: 1, pet_type: "Unicorn" } // matches PetByAge, doesn't care about unicorn not being a type of pet
```

so yes! go ahead and merge if you're happy